### DISCUSSION TITLE
Markdown Support

### DISCUSSION THREAD

**Discussion by Skilly146** [Tsangares/stoned_ape_tools#27] [Feature Requests]
**This is a stretch goal. Similar to the firefox post I'm just putting this out there to see what people think**

## What

Add support for standard markdown when using the in-game message box. Markdown is an easy to use formatting language used by many programs, including github. This would also include a preview feature, most likely a toggle on the bottom row near the send button.

## Why

With proper markdown support it would be much easier to organize longer messages via headings, ordered and unordered lists, italics and bolding, block quotes, horizontal rules, and hyperlinks. It would go well with the existing image preview feature since that uses markdown.

## How

If we assume that all members of a chat are using SAT this feature would be easy to add. We can add a markdown parser like [this JS based one](https://github.com/markdown-it/markdown-it). Then the extension just automatically parses messages before opening and boom markdown support. The challenge would be getting it to work for non-SAT users. Depending on how the messaging system works it should be possible to format the text before sending it, similar to [this firefox plugin](https://addons.mozilla.org/en-US/firefox/addon/markdown-here/) ([Chrome version](https://chrome.google.com/webstore/detail/markdown-here/elifhakcjgalahccnjkneoccemfahfoa)). This plugin formats markdown text in emails before sending them so that any viewer will see the formatting.

## Similar Tools

- [Markdown-It](https://github.com/markdown-it/markdown-it), markdown parser linked above: This is an open source javascript based markdown parser that is used by most major markdown programs and has safety features like disabling html by default.
- [Markdown Here](https://chrome.google.com/webstore/detail/markdown-here/elifhakcjgalahccnjkneoccemfahfoa), markdown chrome extension linked above. This extension can convert markdown formatted text written directly into email websites like gmail

**Comment by Tsangares** [comment]
Oh this is a decent idea. I think maybe not all markdown is the best though. Just headers, bold and italics?

**Reply by Tsangares** [reply]
Because I don't want hyperlinks in chat that can lead to spam or IP tracking and hacking. It is a breach of privacy that can just be avoided by stopping any hyperlinks. The most I would do is the images I just added but even then I restricted the domains to imgbb and imgur using a regex expression that is pretty limited.
